sim/shaman/flame_shock.go

```diff
@@ -119,15 +119,15 @@ func (shaman *Shaman) newFlameShockSpell(rank int, shockTimer *core.Timer) core.
 			spell.DealDamage(sim, result)
 			if result.Landed() {
 				spell.Dot(target).Apply(sim)
-				// shaman.FlameShockDots[rank].Cast(sim, target)
-
 				if shaman.HasRune(proto.ShamanRune_RuneLegsAncestralGuidance) {
 					shaman.lastFlameShockTarget = target
 				}
 
 				if hasPowerSurgeRune && sim.Proc(ShamanPowerSurgeProcChance, "Power Surge Proc") {
 					shaman.PowerSurgeAura.Activate(sim)
 				}
+
+				target = sim.Environment.NextTargetUnit(target)
```

Comment:
I think the way I would do this is actually apply the DoT on line 121 to `result.Target` instead of `target`, because that should be the correct target to apply the DoT to